k8s/c2d_deployer.Makefile

```diff
@@ -62,44 +74,55 @@ app/build:: .build/$(APP_ID)/VERSION \
 	@touch "$@"
 
 
-.PHONY: $(APP_ID)
-$(APP_ID): .build/var/REGISTRY \
-           .build/var/TRACK \
-           .build/var/RELEASE \
-           | .build/$(APP_ID)
-	docker pull $(image-$@)
-	docker tag $(image-$@) "$(REGISTRY)/$@:$(TRACK)"
-	docker tag $(image-$@) "$(REGISTRY)/$@:$(RELEASE)"
-	docker push "$(REGISTRY)/$@:$(TRACK)"
-	docker push "$(REGISTRY)/$@:$(RELEASE)"
-
-
-$(TARGET_IMAGES): .build/var/REGISTRY \
-                  .build/var/TRACK \
-                  .build/var/RELEASE \
-                  | .build/$(APP_ID)
-	docker pull $(image-$@)
-	docker tag $(image-$@) "$(REGISTRY)/$(APP_ID)/$@:$(TRACK)"
-	docker tag $(image-$@) "$(REGISTRY)/$(APP_ID)/$@:$(RELEASE)"
-	docker push "$(REGISTRY)/$(APP_ID)/$@:$(TRACK)"
-	docker push "$(REGISTRY)/$(APP_ID)/$@:$(RELEASE)"
-
-
-.build/$(APP_ID)/tester: .build/var/TESTER_IMAGE \
-                         $(shell find apptest -type f) \
-                         | .build/$(APP_ID)
+.build/$(CHART_NAME)/$(CHART_NAME): .build/var/REGISTRY \
+                                    .build/var/TRACK \
+                                    .build/var/RELEASE \
+                                    | .build/$(CHART_NAME)
+	$(call print_target,$@)
+	docker pull $(image-$(CHART_NAME))
+	docker tag $(image-$(CHART_NAME)) "$(REGISTRY)/$(APP_ID):$(TRACK)"
+	docker tag $(image-$(CHART_NAME)) "$(REGISTRY)/$(APP_ID):$(RELEASE)"
+	docker push "$(REGISTRY)/$(APP_ID):$(TRACK)"
+	docker push "$(REGISTRY)/$(APP_ID):$(RELEASE)"
+	@touch "$@"
+
+
+# map every element of ADDITIONAL_IMAGES list
```

Comment:
qq: Does it mean that `ADDITIONAL_IMAGES` var is required?

Reply:
No, if variable not defined or is empty, then it will also be and empty list so the rule will do nothing